File: app/src/main/java/com/norvera/confession/utils/InjectorUtils.java
package com.norvera.confession.utils;

import android.content.Context;

import com.norvera.confession.data.AppDatabase;
import com.norvera.confession.data.AppRepository;
import com.norvera.confession.data.dao.ExaminationDao;
import com.norvera.confession.ui.main.MainViewModelFactory;

import java.util.concurrent.Executors;


public class InjectorUtils {

    //  REPOSITORY
    public static AppRepository provideAppRepository(AppDatabase appDatabase, AppExecutor executor) {
        return new AppRepository(appDatabase, executor);
    }

    // EXECUTOR
    public static AppExecutor provideAppExecutor() {
        return new AppExecutor(Executors.newSingleThreadExecutor(), new MainThreadExecutor());
    }

    // Database
    public static AppDatabase provideDatabase(Context context) {
       return AppDatabase.getInstance(context);
    }

    public static ExaminationDao provideExaminationDao(Context context) {
        return provideDatabase(context).examinationDao();
    }

    public static ExaminationDao provideCommandmentDao(Context context) {
        return provideDatabase(context).examinationDao();
    }


    public static MainViewModelFactory provideViewModelFactory(Context context) {
        return MainViewModelFactory.getInstance(context);
    }
}

File: app/src/main/java/com/norvera/confession/utils/InjectorUtils.kt
package com.norvera.confession.utils

import android.content.Context

import com.norvera.confession.data.AppDatabase
import com.norvera.confession.data.AppRepository
import com.norvera.confession.data.dao.ExaminationDao
import com.norvera.confession.ui.main.MainViewModelFactory

import java.util.concurrent.Executors


object InjectorUtils {


    //  REPOSITORY
    fun provideAppRepository(appDatabase: AppDatabase, executor: AppExecutor): AppRepository {
        return AppRepository(appDatabase, executor)
    }

    // EXECUTOR
    fun provideAppExecutor(): AppExecutor {
        return AppExecutor(Executors.newSingleThreadExecutor(), MainThreadExecutor())
    }

    // Database
    fun provideDatabase(context: Context): AppDatabase {
        return AppDatabase.getInstance(context)!!
    }

    fun provideExaminationDao(context: Context): ExaminationDao {
        return provideDatabase(context).examinationDao()
    }

    fun provideCommandmentDao(context: Context): ExaminationDao {
        return provideDatabase(context).examinationDao()
    }


    fun provideViewModelFactory(context: Context): MainViewModelFactory {
        return MainViewModelFactory.getInstance(context)!!
    }
}